### DISCUSSION TITLE
"Reupload" files that now match the attributes file in lfs

### DISCUSSION THREAD

**Discussion by VasilisThePikachu** [git-lfs/git-lfs#5632] [General] [answered]
Since I'm slowly adding to my .gitattributes. There some files that I missed and were uploaded with normal git instead of in lfs

Is there a way to "reupload" these files as lfs? Thank you

**Comment by bk2204** [comment, marked as answer]
Hey,

The best way to do this is `git add --renormalize .` and then commit.  You need to do that every time you change `.gitattributes`; if you haven't done that for some previous commits, then you need to know that if you ever check those commits out again, you'll end up with permanently modified files as mentioned in the FAQ.

Once you've made that change, `git push` or `git lfs push` should push the data.

**Comment by chrisd8088** [comment]
You might also want to take a look at the `git lfs migrate import` command.  In my experience, you can rerun this command more than once, if you want to add convert existing files in your Git history to Git LFS objects.  For instance, if you've already got a `.gitattributes` file which looks like:
```
*.dat filter=lfs diff=lfs merge=lfs -text
```

and you discover you also want all `*.bin` files to be Git LFS objects too, you could run:
```
$ git lfs migrate import --include="*.bin"
```

and they should be converted throughout your Git history, and the following will be appended to your `.gitattributes` files:
```
*.bin filter=lfs diff=lfs merge=lfs -text
```

Note that this will rewrite your Git history, creating new commits with new IDs, and so you would then need to force-push your full history to replace the existing one on any remotes.  See the `git-lfs-migrate(1)` manual [page](https://github.com/git-lfs/git-lfs/blob/82f58ad228ff49abb31618152b1992820bb84f5e/docs/man/git-lfs-migrate.adoc) for more details, if this is helpful.
